audbackend/core/backend/minio.py

```diff
@@ -0,0 +1,402 @@
+            finally:
+                response.close()
+                response.release_conn()
+
+    def _ls(
+        self,
+        path: str,
+    ) -> typing.List[str]:
+        r"""List all files under sub-path."""
+        path = self.path(path)
+        objects = self._client.list_objects(
+            self.repository,
+            prefix=path,
+            recursive=True,
+        )
+        paths = [self._collapse(obj.object_name) for obj in objects]
+
+        return paths
+
+    def _move_file(
+        self,
+        src_path: str,
+        dst_path: str,
+        verbose: bool,
+    ):
+        r"""Move file on backend."""
+        self._copy_file(src_path, dst_path, verbose)
+        self._remove_file(src_path)
+
+    def _open(
+        self,
+    ):
+        r"""Open connection to backend."""
+        # At the moment, session management is handled automatically.
+        # If we want to manage this ourselves,
+        # we need to use the `http_client` argument of `minio.Minio`
+        if not self._client.bucket_exists(self.repository):
+            utils.raise_file_not_found_error(self.repository)
+
+    def _owner(
+        self,
+        path: str,
+    ) -> str:
+        r"""Get owner of file on backend."""
+        path = self.path(path)
+        # TODO: owner seems to be empty,
+        # need to check if we have to manage this ourselves?
+        owner = self._client.stat_object(self.repository, path).owner_name
+        return owner
+
+    def path(
+        self,
+        path: str,
+    ) -> str:
+        r"""Convert to backend path.
+
+        Args:
+            path: path on backend
+
+        Returns:
+            path
+
+        """
+        path = path.replace(self.sep, "/")
+        if path.startswith("/"):
+            # /path -> path
+            path = path[1:]
+        return path
+
+    def _put_file(
+        self,
+        src_path: str,
+        dst_path: str,
+        checksum: str,
+        verbose: bool,
+    ):
+        r"""Put file to backend."""
+        dst_path = self.path(dst_path)
+        if verbose:  # pragma: no cover
+            desc = audeer.format_display_message(
+                f"Deploy {src_path}",
+                pbar=False,
+            )
+            print(desc, end="\r")
+
+        content_type = mimetypes.guess_type(src_path)[0] or "application/octet-stream"
+        self._client.fput_object(
+            self.repository,
+            dst_path,
+            src_path,
+            content_type=content_type,
+        )
+
+        if verbose:  # pragma: no cover
+            # Clear progress line
+            print(audeer.format_display_message(" ", pbar=False), end="\r")
+
+    def _remove_file(
+        self,
+        path: str,
+    ):
+        r"""Remove file from backend."""
+        path = self.path(path)
+        self._client.remove_object(self.repository, path)
+
+    def _size(
+        self,
+        path: str,
+    ) -> str:
```

Comment:
This is used by `_copy_file` where it is used to determine whether file size exceeds 5GB.  However I can see no cast in `_copy_file`. Should the return type hint be integer?

Reply:
Good catch, it returns indeed an integer. I corrected the typing information.